packages/timeline-state-resolver/src/integrations/vizMSE/index.ts

```diff
@@ -286,14 +298,18 @@ export class VizMSEDevice extends DeviceWithState<VizMSEState, DeviceOptionsVizM
 							state.layer[layerName] = literal<VizMSEStateLayerInitializeShows>({
 								timelineObjId: l.id,
 								contentType: TimelineContentTypeVizMSE.INITIALIZE_SHOWS,
-								showIds: l.content.showIds,
+								showIds: _.compact(
+									l.content.showNames.map((showName) => this._vizmseManager?.resolveShowNameToId(showName))
```

Comment:
Not sure if my suggestion compiles, but if it does it could be nice to write it like this:
`l.content.showNames.map(this._vizmseManager?.resolveShowNameToId)`

Reply:
Does not due to two reasons. It needs to be bound, and map does not accept `undefined`. That means we end up with
``` typescript
showIds: _.compact(l.content.showNames.map(this._vizmseManager!.resolveShowNameToId.bind(this))),
```
or
``` typescript
showIds: this._vizmseManager
    ? _.compact(l.content.showNames.map(this._vizmseManager.resolveShowNameToId.bind(this)))
    : [],
```
Technically, `this._vizmseManager` should be defined when `convertStateToVizMSE` is called, because `convertStateToVizMSE` and `handleState` (the only method calling the former, that already checks `this._vizmseManager`) are synchronous, but I'm not a fan of slipping in `!`, and I'm not a fan of throwing or bailing out in `convertStateToVizMSE`. Until TS is capable of infering that `this._vizmseManager`, or until we're able to make hints about it, I'll go with the second way of refactoring it because IMO it's ever so slightly better than the original.

Reply:
And that did not work... It should have been 
``` typescript
showIds: this._vizmseManager
    ? _.compact(
        l.content.showNames.map(this._vizmseManager.resolveShowNameToId.bind(this._vizmseManager))
      )
    : [],
```
Then I think I'll go with the original thing because it is shorter and slightly less confusing to my eye

Reply:
Allright, thanks for trying :)